apiserver/src/auth/authorizor.rs

```diff
@@ -139,14 +139,16 @@ impl<T: TokenReviewer> K8STokenAuthorizor<T> {
         token_review_status: &TokenReviewStatus,
         node_selector: &BottlerocketShadowSelector,
     ) -> Result<(), AuthorizationError> {
-        let pod_name = token_review_status
+        let pod_names = token_review_status
             .user
             .as_ref()
             .and_then(|user| user.extra.as_ref())
             .and_then(|extra| extra.get(POD_NAME_INFO_KEY))
-            .and_then(|pod_names| pod_names.first())
             .context(TokenReviewMissingPodName)?;
 
+        snafu::ensure!(pod_names.len() <= 1, TokenReviewMultiplePodNames);
+        let pod_name = pod_names.first().context(TokenReviewMissingPodName)?;
```

Comment:
If I'm not misunderstanding, you can just check that `pod_names.len()` is 1, since you error out when there are more than one, or if there aren't any.

```suggestion
        snafu::ensure!(pod_names.len() == 1, TokenReviewUnexpectedPodNames);
```

Reply:
My problem was that I wanted `TokenReviewMissingPodName` on <1 and `TokenReviewMultiplePodNames` on >1. I suppose I could also solve this with a match, but I'd still want to call `pod_names.first()` anyways.

Reply:
It might be better if you could add some comments here to describe what you are trying to match. It is very confusing on the first read.